[BugFix] Solve issues-3470: Yarn webui fails to obtain task status when submitting a Flink task after turning on Kerberos authentication.

miaoze8 opened the pull request:
解决yarn webui开启Kerberos认证后递交Flink任务无法获取任务状态的问题，增加调用Flink HTTP API时候进行Kerberos认证，
相关issues: https://github.com/DataLinkDC/dinky/issues/3470

## Purpose of the pull request

<!--(For example: This pull request adds spotless plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add spotless-maven-plugin to root pom.xml*
-->
## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
  - *Added dinky-core tests for end-to-end.*
  - *Added UDFUtilTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

Commit:
解决yarn webui开启Kerberos认证后递交Flink任务无法获取任务状态的问题，增加调用Flink HTTP API时候进行Kerberos认证，相关issues: https://github.com/DataLinkDC/dinky/issues/3470

zackyoungh commented:
请使用spotless 格式化代码,详细查看此文：https://www.dinky.org.cn/docs/next/developer_guide/contribution/pull_request#pull-request-code-style

Commit:
解决yarn webui开启Kerberos认证后递交Flink任务无法获取任务状态的问题，增加调用Flink HTTP API时候进行Kerberos认证，相关issues: https://github.com/DataLinkDC/dinky/issues/3470

miaoze8 reviewed (commented):
已用spotless 格式化代码

Commit:
解决yarn webui开启Kerberos认证后递交Flink任务无法获取任务状态的问题，增加调用Flink HTTP API时候进行Kerberos认证，相关issues: https://github.com/DataLinkDC/dinky/issues/3470

miaoze8 reviewed (commented):
已调整

Commit:
解决yarn webui开启Kerberos认证后递交Flink任务无法获取任务状态的问题，增加调用Flink HTTP API时候进行Kerberos认证，相关issues: https://github.com/DataLinkDC/dinky/issues/3470

miaoze8 reviewed (commented):
调整 import x.*

aiwenmo commented on dinky-gateway/src/main/java/org/dinky/gateway/yarn/YarnGateway.java:
```diff
@@ -106,6 +111,8 @@ public abstract class YarnGateway extends AbstractGateway {
     protected YarnClient yarnClient;
 
+    private static boolean ENABLE_KERBEROS_AUTH = false;
+
```
Some clusters require Kerberos authentication, while others do not. It should be determined based on the cluster configuration instead of being uniformly set.

aiwenmo reviewed (commented):
Hi. Thank you for your contribution. I have a question.

actions-user committed:
Merge branch 'dev' of https://github.com/DataLinkDC/dinky into dev

actions-user committed:
Merge branch 'dev' of https://github.com/DataLinkDC/dinky into dev

actions-user committed:
Merge branch 'dev' of https://github.com/DataLinkDC/dinky into dev

Commit:
更改为调用yarn webui时候如HTTP返回401时候,后面补充采用Kerberos方式认证HTTP,获取Flink任务状态
增加调用Flink HTTP API时候进行Kerberos认证，相关issues: https://github.com/DataLinkDC/dinky/issues/3470

Commit:
Merge branch 'dev' of https://github.com/miaoze8/dinky into dev

miaoze8 reviewed (commented):
更改判断方式，仅在获取flink任务状态时候HTTP如果返回401状态时，额外采用Kerberos方式认证HTTP获取flink任务状态

aiwenmo commented on dinky-gateway/src/main/java/org/dinky/gateway/utils/RequestKerberosUrlUtils.java:
```diff
@@ -0,0 +1,158 @@
+        this.principal = principal;
+        this.keyTabLocation = keyTabLocation;
+    }
+
+    public RequestKerberosUrlUtils(String principal, String keyTabLocation, boolean isDebug) {
+        this(principal, keyTabLocation);
+        if (isDebug) {
+            System.setProperty("sun.security.spnego.debug", "true");
+            System.setProperty("sun.security.krb5.debug", "true");
+        }
+    }
+
```
Under what circumstances will the isDebug parameter be true? Currently, only the use of false is in the code.

miaoze8 replied:
这个一般用不到，例如开发时候Kerberos认证失败普通报错找不到原因时候，才用这个参数开启kerberos debug日志进一步找问题，目前只用false，不需要开启true。

aiwenmo commented:
Hi. Run 'mvn spotless:apply' to fix these violations.

Commit:
spotless apply
解决yarn webui开启Kerberos认证后递交Flink任务无法获取任务状态的问题，增加调用Flink HTTP API时候进行Kerberos认证，相关issues: https://github.com/DataLinkDC/dinky/issues/3470

miaoze8 reviewed (commented):
spotless:apply done

aiwenmo reviewed (approved):
LGTM
